src/server.py

```diff
@@ -45,7 +54,10 @@ def _build_pipeline(self):
 
     def __bus_message_cb(self, unused_bus, message):
         if message.type == Gst.MessageType.EOS:
-            self.emit("eos")
+            if self.loop:
```

Comment:
This implementation has the sound loop from end to beginning until you stop it, at which point it stops abruptly. We should check with David whether that is the API we want. Sometimes samples have "loop points" defined, where if you reach the loop endpoint it loops back to the loop startpoint (which is not necessarily the beginning.) Then when you "release" the loop it plays until the end of the sample. Another option is to provide a "fade out" method as an alternative to "stop".

Reply:
We will still have to figure out something for fading the loops in and out, but I will merge this in the meantime.